jsm.go

```diff
@@ -259,34 +259,57 @@ type consumerResponse struct {
 
 // AddConsumer will add a JetStream consumer.
 func (js *js) AddConsumer(stream string, cfg *ConsumerConfig, opts ...JSOpt) (*ConsumerInfo, error) {
-	if cfg != nil && cfg.Durable != _EMPTY_ {
-		consInfo, err := js.ConsumerInfo(stream, cfg.Durable)
+	if cfg == nil {
+		cfg = &ConsumerConfig{}
+	}
+	if cfg.Name != _EMPTY_ && !js.nc.serverMinVersion(2, 9, 0) {
+		return nil, fmt.Errorf("%w: %s", ErrRequireServerVersion, "consumer name requires at least server version 2.9.0")
```

Comment:
not for this PR maybe but we should be doing this style of error wrapping with the `ErrRequireServerVersion` in other places too like: https://github.com/nats-io/nats.go/blob/cc189da40f83a04fa519b36767d496f435b79a85/kv.go#L309